File: android/core/domain/src/main/java/com/enterprise/core/domain/interfaces/PermissionService.kt
package com.enterprise.core.domain.interfaces

/** OS permission request interface with rationale support. */
interface PermissionService {
    suspend fun requestContactsPermission(): PermissionResult
    suspend fun requestCameraPermission(): PermissionResult
    suspend fun requestMicrophonePermission(): PermissionResult
    suspend fun requestNotificationPermission(): PermissionResult
    fun hasPermission(permission: AppPermission): Boolean
}

enum class AppPermission {
    CONTACTS, CAMERA, MICROPHONE, NOTIFICATIONS
}

sealed class PermissionResult {
    data object Granted : PermissionResult()
    data object Denied : PermissionResult()
    data object PermanentlyDenied : PermissionResult()
}

File: ios/EnterpriseMobileApp/Sources/Core/Domain/Interfaces/PermissionService.swift
import Foundation

/// Interface for OS permission requests.
public enum PermissionStatus: Equatable, Sendable {
    case notDetermined
    case authorized
    case denied
    case restricted
}

public enum PermissionType: Equatable, Sendable {
    case contacts
    case camera
    case microphone
    case notifications
}

public protocol PermissionService: Sendable {
    func checkStatus(for permission: PermissionType) async -> PermissionStatus
    func request(_ permission: PermissionType) async -> PermissionStatus
}
